consensus/milestone_tracker/milestone_tracker.c

```diff
@@ -68,11 +68,12 @@ static retcode_t validate_milestone(milestone_tracker_t* const mt,
         memcmp(tx1->hash, candidate->hash, FLEX_TRIT_SIZE_243) != 0) {
       goto done;
     }
-    if ((tx2 = (iota_transaction_t)utarray_eltptr(bundle, 1)) == NULL) {
+    if ((tx2 = (iota_transaction_t)utarray_eltptr(bundle, 1)) == NULL ||
+        memcmp(tx1->branch, tx2->trunk, FLEX_TRIT_SIZE_243) != 0 ||
+        memcmp(tx1->bundle, tx2->bundle, FLEX_TRIT_SIZE_243) != 0) {
```

Comment:
Last condition is not necessary, you call `bundle_validate` which validate transactions from `load_bundle_transactions`, all with same bundle hash, it is redundant in IRI as well